sdk/src/types/block/protocol.rs

```diff
@@ -160,14 +138,9 @@ impl ProtocolParameters {
         network_name_to_id(&self.network_name)
     }
 
-    /// Returns the mana decay factors slice of the [`ProtocolParameters`].
-    pub fn mana_decay_factors(&self) -> &[u32] {
-        &self.mana_decay_factors
-    }
-
-    /// Returns the mana decay factor for the given epoch index.
-    pub fn mana_decay_factor_at(&self, epoch_index: EpochIndex) -> Option<u32> {
-        self.mana_decay_factors.get(*epoch_index as usize).copied()
+    /// Returns the parameters used for mana calculations.
+    pub fn mana_structure(&self) -> &ManaStructure {
```

Comment:
dont we get this from the #[getset(get_copy = "pub")] ?

Reply:
otherwise, i htink were missing a few getters in ManaStructure

Reply:
No, the type is not `Copy` so it is skipped.